src/codegen.cpp

```diff
@@ -10560,11 +10565,11 @@ static Error check_cache(CodeGen *g, Buf *manifest_dir, Buf *digest) {
     cache_list_of_str(ch, g->lib_dirs.items, g->lib_dirs.length);
     cache_list_of_str(ch, g->framework_dirs.items, g->framework_dirs.length);
     if (g->libc) {
-        cache_str(ch, g->libc->include_dir);
-        cache_str(ch, g->libc->sys_include_dir);
-        cache_str(ch, g->libc->crt_dir);
-        cache_str(ch, g->libc->msvc_lib_dir);
-        cache_str(ch, g->libc->kernel32_lib_dir);
```

Comment:
These need the separator, otherwise `a/b/` `c/d` would hash the same as `a/b` `/c/d`.

Reply:
ooo interesting. that that mean that `cache_mem` should also mix in the length?

Reply:
I believe `cache_mem` is used internally as "put these bytes into the hash", whereas the public API functions are more along the lines of "put this thing in the hash and make sure it's separate"

Reply:
Then perhaps that belongs in the `cache_slice` function below and hence answers you question there :)